src/bf_dpu_update.py

```diff
@@ -990,12 +993,70 @@ def factory_reset_bios(self):
         self._wait_for_system_power_on(50, 100)
         print()
 
+    def update_oem_fru(self):
+        """
+        Update the OEM FRU data with the provided key-value pairs in the format 'Section:Key=Value'
+        """
+
+        # Define the required fields for the OEM FRU data
+        required_fields = {
+            'Product:Manufacturer',
+            'Product:SerialNumber',
+            'Product:PartNumber',
+            'Product:Version',
+            'Product:Extra',
+            'Product:ManufactureDate',
+            'Product:AssetTag',
+            'Product:GUID'
+        }
+
+        oem_fru_dict = {}
+        provided_fields = set()
+        if self.debug:
+            print("OEM FRU data to be updated:", self.oem_fru)
+        # Process each item in the provided OEM FRU data
+        for item in self.oem_fru:
+            try:
+                section_key, value = item.split('=')
+                section, key = section_key.split(':')
+                combined_key = section + key
+                oem_fru_dict[combined_key] = value
+                provided_fields.add(section_key)
+                if self.debug:
+                    print(f"Updated FRU field: {section_key} with value: {value}")
+            except ValueError:
+                raise ValueError(f"Invalid format for OEM FRU data: {item}. Expected format 'Section:Key=Value'.")
+
+        # Process each item in the provided OEM FRU data
+        missing_fields = required_fields - provided_fields
+        if missing_fields:
+            raise ValueError(f"Missing required OEM FRU fields: {', '.join(missing_fields)}")
+
+        print("OEM FRU data to be updated:", json.dumps(oem_fru_dict, indent=4))
+
+        # Construct the URL for the HTTP PUT request
+        url = self._get_url_base() + '/Systems/Bluefield/Oem/Nvidia'
+        headers = {'Content-Type': 'application/json'}
+
+        try:
+            # Send the HTTP PUT request to update the OEM FRU data
+            response = self._http_put(url, data=json.dumps(oem_fru_dict), headers=headers)
+            if response.status_code != 200:
+                raise Exception(f"Failed to update OEM FRU data, status code: {response.status_code}")
+            print("OEM FRU data updated successfully.")
+        except Exception as e:
+            print(f"An error occurred while updating OEM FRU data: {e}")
+            return False
+
+        return True
```

Comment:
As stated above,

We use exception to handle error/fail case. So we do not need to return True/False

Reply:
Done